dlmisc.c:95:7: error: 'snd_plugin_dir_set' undeclared (first use in this function)
   95 |  if (!snd_plugin_dir_set) {
      |       ^~~~~~~~~~~~~~~~~~
dlmisc.c:95:7: note: each undeclared identifier is reported only once for each function it appears in
dlmisc.c:98:4: error: 'snd_plugin_dir' undeclared (first use in this function)
   98 |    snd_plugin_dir = strdup(env);
      |

--- a/src/dlmisc.c
+++ b/src/dlmisc.c
@@ -91,6 +91,7 @@
  */
 int snd_dlpath(char *path, size_t path_len, const char *name)
 {
+#ifdef DL_ORIGIN_AVAILABLE
 	snd_dlpath_lock();
 	if (!snd_plugin_dir_set) {
 		const char *env = getenv("ALSA_PLUGIN_DIR");
@@ -104,6 +105,7 @@
 	snprintf(path, path_len, "%s/%s",
 		 snd_plugin_dir ? snd_plugin_dir : ALSA_PLUGIN_DIR, name);
 	snd_dlpath_unlock();
+#endif
 	return 0;
 }
 
